osu.Game/Overlays/KeyBinding/KeyBindingRow.cs

```diff
@@ -118,6 +125,24 @@ private void load(OsuColour colours)
                 buttons.Add(new KeyButton(b));
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            IsDefault.BindValueChanged(resetButtons =>
+            {
+                if (resetButtons.NewValue && !computeIsDefaultValue())
```

Comment:
What is the `computeIsDefaultValue` call here for? Feels like this needs an inline comment if there's a reason for it to exist.

Reply:
This is the awkward part caused by the bool bindable stuff I talked about. In short, this bindable's can change to `true` in two cases:

* User clicks restore value button to restore binding
* User manually changes binding from non-default to default

This check is supposed to prevent the restore/finalise chain from running in case 2, while still keeping case 1 working. I suppose it's not strictly needed, because the restore would do nothing anyways, but that's also a potential argument for keeping it (to avoid busy work).

If the restore value button just fired a plain action this would not be necessary. The keybinding-button value synchronisation would still have to remain in some capacity.

Reply:
Right. I'm not sure how I feel about this being a bindable, since it looks like it's being updated at two very different locations. I'm fine with the bindable being used to convey *outwards* to the restore default button (ie. change the `public` exposed bindable to `IBindable`) but the actual action should probably be a separate action coming in.

Reply:
I'm not sure what you mean by this? Doesn't the RestorableKeyBindingRow need a Bindable?

Also if you are just concerned with this awkward part and not that concerned about using a BindableBool in KeyBindingRow, we could just make the ResetButton action use the IsDefault property and make the RestoreDefaults method private (though that would make the code more unreadable) `k.KeyBindingRow.IsDefault.Value = true`

Reply:
Basically I'm saying the flow of information (whether it is default or not) is fine to be left using bindable flow, but the *set* operation should likely not be done using the same bindable.

Rationale is that it's set in two different places.

Reply:
What I was loosely thinking is that the default value button could have a publicly-declared `Action` attached, that in the 99% defaults to resetting `Current`, and would be overridden here to manually reset the row via a method call or otherwise. Haven't gotten around to trying it yet, though.

Reply:
@sleepyswords are you okay to implement this change?

Reply:
Yeah sure, I'll make the Restore button implement an OsuButton,

Do you still want me to use an IBindable for the KeyBindingRow externally?